extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java

```diff
@@ -938,9 +938,39 @@ public Response doSetContentCreator( @PathParam ( "pathId" ) String pathId, Repo
       @ResponseCode ( code = 500, condition = "Unable to update locale properties due to some other error." ) } )
   public Response doSetLocaleProperties( @PathParam ( "pathId" ) String pathId, @QueryParam ( "locale" ) String locale,
                                          List<StringKeyStringValueDto> properties ) {
+    boolean success = true;
+    String responseMsg = "";
     try {
+      /*
+       [BI-14087] If one of the properties is to change the file/folder name, it is also necessary to run this command to
+       get the file's path to update as well. We need to remove it from the properties, because it can affect future
+       renames. We need to keep it consistent.
+      */
+      String[] pathSplit = pathId.split( ":" );
+      boolean isFileTitleFound = false;
```

Comment:
Why do we need this boolean flag? If the file title isn't found this variable will be false and `fileTitle` will be null.
Validation can be done by evaluating if `fileTitle` is null or not.